### PR TITLE
Morello unwind fixes

### PR DESCRIPTION
**Pull request by bsdjhb**

### CODE REVIEW

**Comment by jrtc27** [comment]
You need to:
1. Make getRegister accept the new range when purecap
2. Make getRegister clear the metadata when asking for an X register (currently it gives the full cap, which is technically wrong)
Similarly 1 for setRegister; unclear if it's worth forcefully clearing the metadata when setting registers or not, given the caller _should_ have passed a plain integer.

**Comment by bsdjhb** [comment]
Oh, bleh, those are true.

File: contrib/subrepo-cheri-libunwind/include/__libunwind_config.h [hunk 1]
```diff
@@ -66,8 +66,8 @@
 #   ifdef __SEH__
 #    error "aarch64 SEH not supported yet"
 #   endif
-#   define _LIBUNWIND_CONTEXT_SIZE 102
-#   define _LIBUNWIND_CURSOR_SIZE 126
+#   define _LIBUNWIND_CONTEXT_SIZE 100
+#   define _LIBUNWIND_CURSOR_SIZE 124
 #   define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_ARM64_CHERI
```
Comment (jrtc27) [inline contrib/subrepo-cheri-libunwind/include/__libunwind_config.h:71]: `_LIBUNWIND_HIGHEST_DWARF_REGISTER_ARM64_CHERI` needs to be changed

Comment (bsdjhb) [reply contrib/subrepo-cheri-libunwind/include/__libunwind_config.h:71]: It is '229' which is the register number of C31.  I think that means it is correct?

Comment (jrtc27) [reply contrib/subrepo-cheri-libunwind/include/__libunwind_config.h:71]: Oh so that was already correct then? Hm.

Comment (bsdjhb) [reply contrib/subrepo-cheri-libunwind/include/__libunwind_config.h:71]: Yes, I'm not sure if I had made that correct and just had not realized I should have used the dwarf numbers for C0-C31, or if it correct from Brett's changes.

**Pull request merged**